### ISSUE TITLE
ISO Writer

### ISSUE THREAD

**Issue by ShalokShalom** [geckolinux/geckolinux-project#20]
Take a look on the ISO Writer here, it offers 2 benefits, compared to SUSEs Writer:

1) You can download and launch it on ANY OS, without packaging and dependency's.
2) You can 'delete' a Stick, which means essentially that the first 10 MB gets overwritten with zeros.
    (That solves the issue with sticks, which are limited to the size of the previously written ISO)

The original source code is from ROSA, they offer already an usable version for all major platforms:
http://wiki.rosalab.com/en/index.php/ROSA_ImageWriter

KaOS fork, patch and rebrand it distro agnostic, i recommend you this version:
https://github.com/KaOSx/isowriter

**Comment by geckolinux** [comment]
Hi there, that's a very interesting find. Thanks for the tip! It looks 
like it's even packaged for openSUSE?
- https://software.opensuse.org/package/imagewriter
- https://software.opensuse.org/package/rosa-imagewriter

Thanks again,
Sb

**Comment by ShalokShalom** [comment]
The first one is that one from suse and the second one is that one from ROSA, i recommend you ISO-Writer from KaOS.

**Comment by ShalokShalom** [comment]
With 1, i mean without packaging it.

**Comment by geckolinux** [comment]
Thanks again for the suggestion @ShalokShalom , the [GeckoLinux wiki](https://github.com/geckolinux/geckolinux-project/wiki#how-to-create-a-bootable-geckolinux-usb-key) has been updated with your suggestion.

Cheers.

**Comment by ShalokShalom** [comment]
ImageUSB as the recommend tool for Windows is great, why did you change that one as well?

It use CheckSums to control if the ISO is correctly written on the stick - the one and only GUI tool i know about, which does that. This control both, the stick and the port and i run frequently into such issues, when i install a Linux based OS for friends.

By the way: I am open for further involvement into your project. 

shalokshalom@protonmail.ch

**Comment by geckolinux** [comment]
Hi again, thanks a lot for offering to help.

The interface of ROSA ImageWriter is cleaner, and it's also nice to have the same tool available for Windows and Linux.

**Comment by ShalokShalom** [comment]
Yeah, thats true. I like a complete other ISO installer at all - much things are improvable in very easy ways.

Users, who comes from Windows and OS X have to learn how to check MD5Sums, access BIOS/UEFI, change boot order (which look and feel frequently different) check both Stick and Port and so on. 

I think its possible to write a simple app, which downloads the ISO via Torrent and offers then 2 options: Write to Stick and to internal drive (HDD/SDD) in the second case add the installer GRUB2 and some specific settings. Then automatic MD5Check and the offer to reboot.

**Comment by geckolinux** [comment]
That sounds similiar to Fedora's [Media Writer](https://fedoraproject.org/wiki/How_to_create_and_use_Live_USB#Quickstart:_Using_Fedora_Media_Writer).

**Comment by ShalokShalom** [comment]
This one downloads the ISO, yeah

There are enough sticks and ports outside, who work fine for media files, since player software comes with error correction and the same sticks are practically broken for the touchy ISO images, where one displaced bit can crash the whole thing in one and the other way. Some users might use a random stick, which isnt tested/used at all since months/years. 

So control the Checksum, once the write process is done. 

Plus: The live boot from HDD/SDD ships the huge benefit, that no one need to access and configure the boot order in their BIOS/UEFI, the users get simply offered the hint, that the installation succeed, once a reboot happen.

So;

1) Download ISO
2) Offer boot from 
    a) internal drive (default, for beginners) 
    b) live stick 
3) Install to media
4) Verify ISO 
(First between supplied Checksum and media, if that one fails between supplied Checksum and downloaded ISO, display corresponding error message plus possible solutions) 

P.S; The installation to internal drives brings also the option to use time zone/language and keyboard layout from Windows and write it as the defaults into the ISO.

**Comment by ShalokShalom** [comment]
Update: The KaOS version of the ROSA Image Writer offers now the mentioned verification:

https://github.com/KaOSx/isowriter

**Comment by geckolinux** [comment]
Very nice, is it available for other distros and/or Windows?

**Comment by ShalokShalom** [comment]
http://doc.qt.io/QtForDeviceCreation/qtee-static-linking.html
